packages/medusa-payment-paypal/src/services/paypal-provider.js

```diff
@@ -284,11 +282,18 @@ class PayPalProviderService extends PaymentService {
   }
 
   /**
-   * Cancels payment for Stripe payment intent.
-   * @param {object} paymentData - payment method data from cart
+   * Cancels payment for paypal payment.
+   * @param {Payment} payment - payment object
    * @returns {Promise<object>} canceled payment intent
    */
   async cancelPayment(payment) {
+    const order = await this.retrievePayment(payment.data)
+    const isAlreadyCancelled = order.status === "VOIDED"
+    const isCancelledAndFullyRefund = order.status === "COMPLETED" && !!order.invoice_id
+    if (isAlreadyCancelled || isCancelledAndFullyRefund) {
+      return order
```

Comment:
nit: should be American English (you had it right the first time :D)

Reply:
😂  I should have sticked with it the first time